src/personnummer.js

```diff
@@ -1,246 +1,229 @@
-/**
- * Compare the two dates and return -1, 0 or 1.
- *
- * @param {Date} dateLeft
- * @param {Date} dateRight
- *
- * @return {numbers}
- */
-const compareAsc = (dateLeft, dateRight) => {
-  const diff = dateLeft.getTime() - dateRight.getTime()
-  return diff < 0 ? -1 : diff > 0 ? 1 : diff;
-};
-
-/**
- * Get the number of full years between the given dates.
- *
- * @param {Date} dateLeft
- * @param {Date} dateRight
- *
- * @return {numbers}
- */
-const diffInYears = (dateLeft, dateRight) => {
-  const sign = compareAsc(dateLeft, dateRight);
-  const yearDiff = Math.abs(dateLeft.getFullYear() - dateRight.getFullYear());
-
-  dateLeft.setFullYear(dateLeft.getFullYear() - sign * yearDiff);
-
-  const isLastYearNotFull = compareAsc(dateLeft, dateRight) === -sign
-  const result = sign * (yearDiff - isLastYearNotFull);
-
-  return result === 0 ? 0 : result;
-};
-
-/**
- * Calculates the Luhn checksum of a string of digits.
- *
- * @param {string|number} str
- *
- * @return {number}
- */
-const luhn = str => {
-  let v = 0;
-  let sum = 0;
-
-  str += '';
-
-  for (var i = 0, l = str.length; i < l; i++) {
-    v = str[i];
-    v *= 2 - i % 2;
-    if (v > 9) {
-      v -= 9;
-    }
-    sum += v;
-  }
-
-  return Math.ceil(sum / 10) * 10 - sum;
-};
-
-/**
- * Test if the input parameters are a valid date or not.
- *
- * @param {int} year
- * @param {int} month
- * @param {int} day
- *
- * @return {boolean}
- */
-const testDate = (year, month, day) => {
-  month -= 1;
-  const date = new Date(year, month, day);
-  return !('' + date.getFullYear() !== year || date.getMonth() !== month || date.getDate() !== day);
-};
-
-/**
- * Parse a Swedish social security number and get the parts.
- *
- * @param {string|int} ssn
- *
- * @return {object}
- */
-const getParts = (ssn) => {
-  ssn += '';
-
-  const reg = /^(\d{2}){0,1}(\d{2})(\d{2})(\d{2})([\-|\+]{0,1})?(\d{3})(\d{0,1})$/;
-  const match = reg.exec(ssn);
-
-  if (!match) {
-    return {};
-  }
+import { PersonnummerError } from './errors';
+import { diffInYears, luhn, testDate } from './utils';
 
-  let century = match[1];
-  let year = match[2];
-  let month = match[3];
-  let day = match[4];
-  let sep = match[5];
-  let num = match[6];
-  let check = match[7];
+class Personnummer {
 
-  if (typeof century === 'undefined' || !century.length) {
-    const d = new Date;
-    let baseYear = 0;
+  /**
+   * Personnummer age.
+   *
+   * @var {string}
+   */
+  age = '';
```

Comment:
Do we want those as public fields? I think it might be best to set them as private (`#age`) (even if the # is ugly... hehe) and use getters for the given property instead:

```
get Age {
  return this.#age;
}
```

Else they could be changed in the runtime, which could create some issues.

Reply:
Yeah, maybe better to use getters even if I don't see any big problems to have it as a it is. v3 spec has this as public fields :)

Reply:
But if we should follow the spec I think we should do this

```
get age {
  return #_age
}
```

or something like that. Most packages in the JS-world use lowercases :)

Reply:
Added getters!